posthog/migrations/0362_auto_20231107_2008.py

```diff
@@ -0,0 +1,22 @@
+# Generated by Django 3.2.19 on 2023-11-07 20:08
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("posthog", "0361_add_plugin_config_ui_fields"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="team",
+            name="external_data_workspace_last_synced",
+            field=models.DateTimeField(blank=True, null=True),
+        ),
+        migrations.AddField(
+            model_name="team",
+            name="external_data_workspace_rows_synced_in_month",
```

Comment:
Current month. I'm using this field to make sure that the limits are respected in real time. The job that will actually provide the usage report for billing can look back for any period since those will just be events emitted like the rest of our event system.

Reply:
> I'm using this field to make sure that the limits are respected in real time.

Curious how this works for this use-case? I don't think any limiting should be based on current month (eg. month od novenber) but maybe I'm misunderstanding your purpose

Reply:
Oh yep fair q. I'm setting it up so that the limit will apply immediately once the number is crossed (to prevent unexpected cost run ups for us). 

So let's say the user sets a limit at 10 rows (converted from whatever dollar amount they set). Each day airbyte syncs 1 row. By November 10th, 10 rows will have been synced. At this point, `external_data_workspace_rows_synced_in_month` will read 10 and so the celery task `check_external_data_source_billing_limit_by_team` will find that the 10 row limit is reached. All the expected jobs for the user will be turned off so that they dont keep running up the rows synced count and we don't incur any extra costs that we would have to pay airbyte.

Reply:
**had a huddle on this. Modifying the implementation to leverage the existing limiting logic in quota_limiting.py!